qiskit/algorithms/fidelities/base_fidelity.py

```diff
@@ -0,0 +1,253 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Base fidelity interface
+"""
+
+from __future__ import annotations
+from abc import ABC, abstractmethod
+from typing import Sequence, Mapping
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import ParameterVector
+
+
+class BaseFidelity(ABC):
```

Comment:
@woodsp-ibm I agree that it would make sense for the base class to define a common interface, but the reason I did not include `run()` in the base class was that I was not sure about how to specify the return type. The base class does not enforce the use of primitives, but the return of `run()` for the `ComputeUncompute` class (former `Fidelity`) is a `PrimitiveJob`... When I had `evaluate()`, instead of `run()`, I did include it in the base class, but this whole sync/async separation makes me think that it would be strange to include the "async" run in an interface that doesn't necessarily enable it. Does this make sense? do you have any suggestions?

Reply:
I think it would make sense to have `run` here since it'll be the same for all fidelity implementations, which then just have to implement `_run` 🙂

Reply:
If we expect all fidelities to be passable to algorithms and consumed consistently whatever methods/properties by which an algorithm does that need to be here if this is the type (interface) that gets passed. In terms of the return type from run() it needs to be defined as something so the algorithm knows what it is - a JobV1, PrimitiveJob, FidelityJob or whatever. You have it as PrimitiveJob in the class, does that not suffice. I saw some comments elsewhere about Jobs, whichever we choose list is as the return type.

Reply:
ok, adding `run()` to the base class and specifying `PrimitiveJob` as the return type for the moment (might become `AlgorithmsJob` if we agree on that route)